islpy/__init__.py

```diff
@@ -145,14 +147,67 @@
 EXPR_CLASSES = tuple(cls for cls in ALL_CLASSES
         if "Aff" in cls.__name__ or "Polynomial" in cls.__name__)
 
-DEFAULT_CONTEXT = Context()
```

Comment:
Done. 
However, this means < 3.7 we don't get a warning.

Reply:
> However, this means < 3.7 we don't get a warning.

That's life. I don't see a way to avoid that. Simply breaking code that runs on 3.6 (which we claim is supported) is not an option, IMO.